src/odemis/gui/comp/overlay/world.py

```diff
@@ -832,6 +833,604 @@ def deactivate(self):
+                "s_l": sx - self.HOVER_MARGIN,
+                "s_r": sx + self.HOVER_MARGIN,
+                "s_t": sy - self.HOVER_MARGIN,
+                "s_b": sy + self.HOVER_MARGIN,
+                "e_l": ex - self.HOVER_MARGIN,
+                "e_r": ex + self.HOVER_MARGIN,
+                "e_t": ey - self.HOVER_MARGIN,
+                "e_b": ey + self.HOVER_MARGIN,
+            })
+
+    def _outer_rect(self, convert_to_buffer=False):
+        """ Return the outer rectangle of the selection (x, y, w, h) """
+        return self._edges_to_rect(self._edges['o_l'], self._edges['o_t'],
+                                   self._edges['o_r'], self._edges['o_b'],
+                                   convert_to_buffer)
+
+    def _start_rect(self, convert_to_buffer=False):
+        """ Return the rectangle of the start position (x, y, w, h) """
+        return self._edges_to_rect(self._edges['s_l'], self._edges['s_t'],
+                                   self._edges['s_r'], self._edges['s_b'],
+                                   convert_to_buffer)
+
+    def _end_rect(self, convert_to_buffer=False):
+        """ Return the rectangle of the end position (x, y, w, h) """
+        return self._edges_to_rect(self._edges['e_l'], self._edges['e_t'],
+                                   self._edges['e_r'], self._edges['e_b'],
+                                   convert_to_buffer)
+
+    def _edges_to_rect(self, x1, y1, x2, y2, convert_to_buffer=False):
+        """ Return a rectangle of the form (x, y, w, h) """
+        if convert_to_buffer:
+            x1, y1 = self.cnvs.view_to_buffer((x1, y1))
+            x2, y2 = self.cnvs.view_to_buffer((x2, y2))
+            return self._points_to_rect(x1, y1, x2, y2)
+        else:
+            return self._points_to_rect(x1, y1, x2, y2)
+
+    @staticmethod
+    def _points_to_rect(left, top, right, bottom):
+        """ Transform two (x, y) points into a (x, y, w, h) rectangle """
+        return left, top, right - left, bottom - top
+
+    def _debug_edges(self, ctx, convert_to_buffer=False):
+
+        if self._edges:
+            outer_rect = self._outer_rect(convert_to_buffer)
+
+            ctx.set_line_width(0.5)
+            ctx.set_dash([])
+
+            ctx.set_source_rgba(0, 0, 1, 1)
+            ctx.rectangle(*outer_rect)
+            ctx.stroke()
+
+            start_rect = self._start_rect(convert_to_buffer)
+            end_rect = self._end_rect(convert_to_buffer)
+
+            ctx.set_source_rgba(0.3, 1, 0.3, 1)
+            ctx.rectangle(*start_rect)
+            ctx.stroke()
+
+            ctx.set_source_rgba(0.6, 1, 0.6, 1)
+            ctx.rectangle(*end_rect)
+            ctx.stroke()
+
+    def get_hover(self, vpos):
+        """
+        Check if the given position is on/near a selection edge or inside the selection.
+        It returns a "gui.HOVER_*"
+        """
+
+        if self._edges:
+
+            if self.pixel_ruler:
+                return gui.HOVER_LINE
```

Comment:
This doesn't make sense. The mouse has to be near the ruler. It cannot happen for *any* position.
=> Check if its within the outer bounding box, and if not, return `HOVER_NONE`.